File: back/pola/src/main/java/com/jinjinjara/pola/data/dto/response/FileDetailResponse.java
package com.jinjinjara.pola.data.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FileDetailResponse {

    @JsonProperty("id")
    private Long id;

    @JsonProperty("user_id")
    private Long userId;

    @JsonProperty("category_id")
    private Long categoryId;

    @JsonProperty("src")
    private String src;

    @JsonProperty("type")
    private String type;

    @JsonProperty("context")
    private String context;

    @JsonProperty("ocr_text")
    private String ocrText;

    @JsonProperty("vector_id")
    private Long vectorId;

    @JsonProperty("file_size")
    private Long fileSize;

    @JsonProperty("share_status")
    private Boolean shareStatus;

    @JsonProperty("favorite")
    private Boolean favorite;

    @JsonProperty("favorite_sort")
    private Integer favoriteSort;

    @JsonProperty("favorited_at")
    private LocalDateTime favoritedAt;

    @JsonProperty("views")
    private Integer views;

    @JsonProperty("platform")
    private String platform;

    @JsonProperty("origin_url")
    private String originUrl;

    @JsonProperty("created_at")
    private LocalDateTime createdAt;

    @JsonProperty("last_viewed_at")
    private LocalDateTime lastViewedAt;

    @JsonProperty("tags")
    private List<TagResponse> tags;
}

File: mobile/Pola/app/src/main/java/com/jinjinjara/pola/data/remote/dto/response/FileDetailResponse.kt
package com.jinjinjara.pola.data.remote.dto.response

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * 파일 상세 정보 응답
 */
@JsonClass(generateAdapter = true)
data class FileDetailResponse(
    @Json(name = "data")
    val data: FileDetailData,
    @Json(name = "message")
    val message: String,
    @Json(name = "status")
    val status: String
)

@JsonClass(generateAdapter = true)
data class FileDetailData(
    @Json(name = "id")
    val id: Long,
    @Json(name = "user_id")
    val userId: Long,
    @Json(name = "category_id")
    val categoryId: Long,
    @Json(name = "src")
    val src: String,
    @Json(name = "type")
    val type: String,
    @Json(name = "context")
    val context: String?,
    @Json(name = "ocr_text")
    val ocrText: String?,
    @Json(name = "vector_id")
    val vectorId: Long?,
    @Json(name = "file_size")
    val fileSize: Long,
    @Json(name = "share_status")
    val shareStatus: Boolean,
    @Json(name = "favorite")
    val favorite: Boolean,
    @Json(name = "favorite_sort")
    val favoriteSort: Int,
    @Json(name = "favorited_at")
    val favoritedAt: String?,
    @Json(name = "views")
    val views: Int,
    @Json(name = "platform")
    val platform: String,
    @Json(name = "origin_url")
    val originUrl: String,
    @Json(name = "created_at")
    val createdAt: String,
    @Json(name = "last_viewed_at")
    val lastViewedAt: String?,
    @Json(name = "tags")
    val tags: List<FileTag>
)

@JsonClass(generateAdapter = true)
data class FileTag(
    @Json(name = "id")
    val id: Long,
    @Json(name = "tagName")
    val tagName: String
)
